bundles/org.openhab.transform.basicprofiles/src/main/java/org/openhab/transform/basicprofiles/internal/profiles/InactivityProfile.java

```diff
@@ -0,0 +1,175 @@
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.thing.profiles.ProfileCallback;
+import org.openhab.core.thing.profiles.ProfileContext;
+import org.openhab.core.thing.profiles.ProfileTypeUID;
+import org.openhab.core.thing.profiles.StateProfile;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.openhab.transform.basicprofiles.internal.config.InactivityProfileConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Changes the state of a {@link Switch} Item depending on whether data has been sent by the
+ * binding during the configured timeout window.
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ */
+@NonNullByDefault
+public class InactivityProfile implements StateProfile, AutoCloseable {
+
+    private static final Duration DEFAULT_TIMEOUT = Duration.ofHours(1);
+    private static final Cleaner CLEANER = Cleaner.create();
+
+    public static final AtomicBoolean DEBUG_CLEANER_TASK_CALLED = new AtomicBoolean();
+
+    private final Logger logger = LoggerFactory.getLogger(InactivityProfile.class);
+
+    private static class CleanerTaskCanceller implements Runnable {
+        private @Nullable ScheduledFuture<?> task;
+
+        public void setTask(@Nullable ScheduledFuture<?> task) {
+            this.task = task;
+        }
+
+        @Override
+        public void run() {
+            DEBUG_CLEANER_TASK_CALLED.set(true);
+            if (task instanceof ScheduledFuture target) {
+                target.cancel(true);
+            }
+        }
+    }
+
+    private final ProfileCallback callback;
+    private final ScheduledExecutorService scheduler;
+    private final Duration timeout;
+    private final boolean inverted;
+    private final CleanerTaskCanceller cleanerTaskCanceller;
+    private boolean closed;
+
+    private @Nullable ScheduledFuture<?> timeoutTask;
+
+    public InactivityProfile(ProfileCallback callback, ProfileContext context) {
+        InactivityProfileConfig config = context.getConfiguration().as(InactivityProfileConfig.class);
+
+        this.callback = callback;
+        this.scheduler = context.getExecutorService();
+        this.inverted = config.inverted != null ? config.inverted : false;
+
+        this.cleanerTaskCanceller = new CleanerTaskCanceller();
+        CLEANER.register(this, cleanerTaskCanceller);
+
+        Duration timeout;
+        try {
+            timeout = parseDuration(config.timeout);
+            logger.debug("Profile created with timeout:{}, inverted:{}", timeout, inverted);
+        } catch (IllegalArgumentException e) {
+            timeout = DEFAULT_TIMEOUT;
+            logger.warn("Profile configuration timeout value \"{}\" is invalid", config.timeout);
+        }
+
+        this.timeout = timeout;
+        onStateUpdateFromHandler(UnDefType.NULL); // dummy to set initial item state
```

Comment:
Should we leave the item's initial state either alone (not set it, so that it's either NULL as it was initialized in GenericItem or restored from persistence). This way, the item doesn't just get an initial "OFF/ON" (Indication that data was recently received). Instead, we know that "we don't know yet so far".